components/script/dom/bindings/construct_hook.rs

```diff
@@ -0,0 +1,59 @@
+// construct_hook.rs
+
+use std::ptr;
+use crate::dom::bindings::htmlconstructor;
+use crate::dom::bindings::utils::{GetConstructorNameForReporting, MakeNativeName};
+use crate::dom::bindings::root::DomRoot;
+use crate::jsapi::{JSContext, JSObject, CallArgs, GlobalScope, SafeJSContext};
+use crate::dom::bindings::import::module::callargs_is_constructing;
+use crate::dom::bindings::import::module::throw_constructor_without_new;
+use crate::dom::bindings::codegen::PrototypeList::ID;
+// Function for HTML constructor case
+pub unsafe fn construct_html_custom(global: &GlobalScope, cx: *mut JSContext, args: &CallArgs, descriptor_name: &str) -> bool {
+    let global = DomRoot::downcast::<dom::types::Global>(global).unwrap(); // Adjust type as needed
+    let prototype_id = match descriptor_name {
+        "HTMLDivElement" => PrototypeList::ID::HTMLDivElement,
+        "HTMLSpanElement" => PrototypeList::ID::HTMLSpanElement,
+        // Add cases for other descriptor names as needed
+        _ => panic!("Unsupported descriptor name: {}", descriptor_name),
+    };
+    htmlconstructor::call_html_constructor::<dom::types::Global>(
```

Comment:
Instead of using `<dom::types::Global>` here, we need to add a generic type to this function (`pub unsafe fn construct_html_custom<T: DerivedFrom<Element> + DomObject>`) and use that generic type (`htmlconstructor::call_html_constructor::<T>`).